searx/engines/dailymotion.py

```diff
@@ -21,23 +26,89 @@
 # engine dependent config
 categories = ['videos']
 paging = True
+number_of_results = 10
+
+time_range_support = True
+time_delta_dict = {
+    "day":  timedelta(days=1),
+    "week": timedelta(days=7),
+    "month": timedelta(days=31),
+    "year": timedelta(days=365),
+}
+
+safesearch = True
+safesearch_params = {2: '&is_created_for_kids=true', 1: '&is_created_for_kids=true', 0: ''}
 
 # search-url
-# see http://www.dailymotion.com/doc/api/obj-video.html
-search_url = 'https://api.dailymotion.com/videos?fields=created_time,title,description,duration,url,thumbnail_360_url,id&sort=relevance&limit=5&page={pageno}&{query}'  # noqa
-supported_languages_url = 'https://api.dailymotion.com/languages'
+# - https://developers.dailymotion.com/tools/
+# - https://www.dailymotion.com/doc/api/obj-video.html
+
+result_fields = [
+    'allow_embed',
+    'description',
+    'title',
+    'created_time',
+    'duration',
+    'url',
+    'thumbnail_360_url',
+    'id',
+]
+search_url = (
+    'https://api.dailymotion.com/videos?'
+    'fields={fields}&password_protected={password_protected}&private={private}&sort={sort}&limit={limit}'
+).format(
+    fields=','.join(result_fields),
+    password_protected= 'false',
+    private='false',
+    sort='relevance',
+    limit=number_of_results,
+)
+iframe_src = "https://www.dailymotion.com/embed/video/{video_id}"
+
+# The request query filters by 'languages' & 'country', therefore instead of
+# fetching only languages we need to fetch locales.
+supported_languages_url = 'https://api.dailymotion.com/locales'
+supported_languages_iso639: Set[str] = set()
+
+
+def init(_engine_settings):
+    global supported_languages_iso639
+    supported_languages_iso639 = set([language.split('_')[0] for language in supported_languages])
 
 
-# do search-request
 def request(query, params):
-    if params['language'] == 'all':
-        locale = 'en-US'
-    else:
-        locale = match_language(params['language'], supported_languages)
 
-    params['url'] = search_url.format(
-        query=urlencode({'search': query, 'localization': locale}), pageno=params['pageno']
-    )
+    if not query:
+        return False
```

Comment:
nit: I would fix this issue in the `searx.search` page: engines should not received an empty query.